### PR TITLE
fix issue of using defaults instead of limits

### PR DESCRIPTION
**Pull request by ckiosidis**
#patch

# TL;DR
_Please replace this text with a description of what this PR accomplishes._

When deploying flyte v0.17.0  we noticed that the values from task default in flyteadmin were used both for pod Requests and Limits.

I haven't tested this PR, and not sure how to test the changes.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_
I haven't tested this

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by jeevb** [comment]
This behavior is intended: https://github.com/flyteorg/flyteadmin/pull/231
